cms/core/blocks/base.py

```diff
@@ -36,20 +42,35 @@ def get_link(self, context: dict | None = None) -> dict[str, str] | None:
             if has_description:
                 value["description"] = desc or getattr(page.specific_deferred, "summary", "")
 
+        if not value:
+            return None
+
+        value["metadata"] = {
```

Comment:
Thinking of this, ideally content type and date would only apply for external URLs as we can get this data from the page instances. Let's leave as is for now

Reply:
Discussed this in Slack. The more I use the block, the more I feel we should pull the type and release date automatically if choosing pages. And only override if the values are manually set

Reply:
Addressed in [0276d51](https://github.com/ONSdigital/dis-wagtail/pull/253/commits/0276d510196c4c55f6918df55bc1ce6d4747cfd5)